src/main/java/org/embulk/input/marketo/delegate/LeadServiceResponseMapperBuilder.java

```diff
@@ -0,0 +1,85 @@
+package org.embulk.input.marketo.delegate;
+
+import com.google.common.base.Optional;
+import org.embulk.base.restclient.ServiceResponseMapper;
+import org.embulk.base.restclient.ServiceResponseMapperBuildable;
+import org.embulk.base.restclient.record.ValueLocator;
+import org.embulk.config.Config;
+import org.embulk.config.ConfigDefault;
+import org.embulk.input.marketo.MarketoService;
+import org.embulk.input.marketo.MarketoUtils;
+import org.embulk.input.marketo.model.MarketoField;
+import org.embulk.spi.Exec;
+import org.slf4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by tai.khuu on 5/21/18.
+ */
+public class LeadServiceResponseMapperBuilder<T extends LeadServiceResponseMapperBuilder.PluginTask> implements ServiceResponseMapperBuildable<T>
+{
+    private static final Logger LOGGER = Exec.getLogger(LeadServiceResponseMapperBuilder.class);
+    private MarketoService marketoService;
+
+    private T pluginTask;
+
+    public interface PluginTask extends MarketoBaseInputPluginDelegate.PluginTask
+    {
+        @Config("included_fields")
+        @ConfigDefault("null")
+        Optional<List<String>> getIncludedLeadFields();
+
+        @Config("extracted_fields")
+        @ConfigDefault("[]")
+        List<String> getExtractedFields();
+
+        void setExtractedFields(List<String> extractedFields);
+    }
+
+    public LeadServiceResponseMapperBuilder(T task, MarketoService marketoService)
+    {
+        this.pluginTask = task;
+        this.marketoService = marketoService;
+    }
+
+    protected List<MarketoField> getLeadColumns()
+    {
+        List<MarketoField> columns = marketoService.describeLead();
+        if (pluginTask.getIncludedLeadFields().isPresent() && !pluginTask.getIncludedLeadFields().get().isEmpty()) {
+            List<MarketoField> filteredColumns = new ArrayList<>();
+            List<String> includedFields = pluginTask.getIncludedLeadFields().get();
```

Comment:
nit: the code is totally okay, but do you think below may improve a little of readability? It can easily say: filter only `columns` which is included by `includedLeadFields`. Anyway, just for discussion.
```java
            final FluentIterable<String> includedFields = FluentIterable.from(pluginTask.getIncludedLeadFields().get());
            return FluentIterable.from(columns).filter(new Predicate<MarketoField>()
            {
                @Override
                public boolean apply(final MarketoField col)
                {
                    Predicate<String> equalsIgnoreCase = new Predicate<String>()
                    {
                        @Override
                        public boolean apply(@Nullable String input)
                        {
                            return col.getName().equalsIgnoreCase(input);
                        }
                    };
                    return includedFields.anyMatch(equalsIgnoreCase);
                }
            }).toList();
```